src/Providers/OpenAI/Handlers/Stream.php

```diff
@@ -50,37 +57,50 @@ public function handle(Request $request): Generator
      */
     protected function processStream(Response $response, Request $request, int $depth = 0): Generator
     {
-        // Prevent infinite recursion with tool calls
-        if ($depth >= $request->maxSteps()) {
-            throw new PrismException('Maximum tool call chain depth exceeded');
-        }
         $text = '';
         $toolCalls = [];
 
         while (! $response->getBody()->eof()) {
             $data = $this->parseNextDataLine($response->getBody());
 
-            // Skip empty data or DONE markers
             if ($data === null) {
                 continue;
             }
 
-            // Process tool calls
+            if (Str::startsWith(data_get($data, 'type', ''), 'response.reasoning')) {
+                $this->pendingReasoningId = data_get($data, 'id')
+                    ?? data_get($data, 'reasoning.id');
+
+                continue;
+            }
+
             if ($this->hasToolCalls($data)) {
                 $toolCalls = $this->extractToolCalls($data, $toolCalls);
 
                 continue;
             }
 
-            // Handle tool call completion
-            if ($this->mapFinishReason($data) === FinishReason::ToolCalls) {
-                yield from $this->handleToolCalls($request, $text, $toolCalls, $depth);
+            if ($this->isFinalEvent($data)) {
+                if ($toolCalls !== []) {
```

Comment:
This should be removed as well since `handleToolCalls` is moved to the end after `while` is done.
I don't think there should be any early return, yield chunks when needed and let stream finish.